data_steward/tools/add_hpo.py

```diff
@@ -313,21 +318,35 @@ def main(hpo_id, org_id, hpo_name, bucket_name, display_order, addition_type,
         This is necessary because a config update will need to be verified in the curation_devops repo
         before updating the lookup tables. Can take values "update_config" or "update_lookup_tables"
     :param hpo_site_mappings_path: path to csv file containing hpo site information
+    :param run_as: Service Account for impersonation
+    :param us_state: Site's PIIState ID in PIIState_XY format
+    :param value_source_concept_id: Value Source Concept ID for the State.
     :return:
     """
+
+    impersonation_creds = get_impersonation_credentials(
+        run_as, target_scopes=CDR_SCOPES)
+    bq_client = BigQueryClient(project_id=project_id,
+                               credentials=impersonation_creds)
+    gcs_client = StorageClient(project_id=project_id,
+                               credentials=impersonation_creds)
+
     if addition_type == "update_config":
         add_hpo_site_mappings_csv(hpo_id, hpo_name, org_id,
                                   hpo_site_mappings_path, display_order)
     elif addition_type == "update_lookup_tables":
-        if bucket_access_configured(bucket_name):
+        if bucket_access_configured(gcs_client, bucket_name):
             LOGGER.info(f'Accessing bucket {bucket_name} successful. '
                         f'Proceeding to add site.')
-            add_lookups(hpo_id, hpo_name, org_id, bucket_name, display_order)
+            check_state_code_format(us_state)
```

Comment:
This can be removed since check can be called in the parser
```suggestion
```

Reply:
oh, yes.. that too. thanks